astropy_helpers/commands/build_sphinx.py

```diff
@@ -26,10 +81,6 @@ class AstropyBuildDocs(SphinxBuildDoc):
     that is built by the setup ``build`` command, rather than whatever is
     installed on the system.  To build docs against the installed version, run
     ``make html`` in the ``astropy/docs`` directory.
-
-    This also automatically creates the docs/_static directories--this is
```

Comment:
You only get a warning from Sphinx if you define ``html_static_path`` but don't use it.

The package template, and by extension a number of affiliated packages don't define ``html_static_path`` by default, so no warning is emitted.

The core package defines ``html_static_path`` and has files in ``_static``.

Basically the previous code was just to catch the case where people were defining ``html_static_path`` and then not actually using it. I don't really think we should be working around this, since it's an easy fix (just don't define ``html_static_path``).

Reply:
While I see your point, this is a subtle incompatibility.  That is, if there's anyone who defined `html_static_path` and counted on this machinery to build their `_static` will now see this breaking their build, and probably not know why (because the point of this stuff was to make it so they don't have to know about it).

So I'd say this should either stay or we should for at least one version create a more helpful warning that tells people exactly what to do.

Reply:
To make sure we only show the warning for users who both have the setting set and the folder missing, and to have the message at the end of the build, I added this as a mini-extension in sphinx-astropy: https://github.com/astropy/sphinx-astropy/pull/13#issuecomment-438760820